homeassistant/components/litterrobot/button.py

```diff
@@ -23,22 +26,64 @@ async def async_setup_entry(
 ) -> None:
     """Set up Litter-Robot cleaner using config entry."""
     hub: LitterRobotHub = hass.data[DOMAIN][entry.entry_id]
-    async_add_entities(
-        LitterRobotResetWasteDrawerButton(
-            robot=robot, entity_type=TYPE_RESET_WASTE_DRAWER, hub=hub
-        )
-        for robot in hub.litter_robots()
-        if isinstance(robot, LitterRobot3)
+    entities: Iterable[LitterRobotButtonEntity] = itertools.chain(
+        (
+            LitterRobotButtonEntity(
+                robot=robot, hub=hub, description=LITTER_ROBOT_BUTTON
+            )
+            for robot in hub.litter_robots()
+            if isinstance(robot, LitterRobot3)
+        ),
+        (
+            LitterRobotButtonEntity(
+                robot=robot, hub=hub, description=FEEDER_ROBOT_BUTTON
+            )
+            for robot in hub.feeder_robots()
+        ),
     )
+    async_add_entities(entities)
+
+
+@dataclass
+class RobotButtonEntityDescription(ButtonEntityDescription, Generic[_RobotT]):
+    """A class that describes robot button entities."""
+
+    press_fn: Callable[[_RobotT], Coroutine[Any, Any, bool] | None] = lambda _: None
+
+
+LITTER_ROBOT_BUTTON = RobotButtonEntityDescription[LitterRobot3](
+    key="reset_waste_drawer",
+    name="Reset Waste Drawer",
+    icon="mdi:delete-variant",
+    entity_category=EntityCategory.CONFIG,
+    press_fn=lambda robot: robot.reset_waste_drawer(),
+)
+FEEDER_ROBOT_BUTTON = RobotButtonEntityDescription[FeederRobot](
+    key="give_snack",
+    name="Give snack",
+    icon="mdi:candy-outline",
+    press_fn=lambda robot: robot.give_snack(),
+)
+
 
+class LitterRobotButtonEntity(LitterRobotEntity[_RobotT], ButtonEntity):
+    """Litter-Robot button entity."""
 
-class LitterRobotResetWasteDrawerButton(LitterRobotEntity[LitterRobot3], ButtonEntity):
-    """Litter-Robot reset waste drawer button."""
+    entity_description: RobotButtonEntityDescription[_RobotT]
 
-    _attr_icon = "mdi:delete-variant"
-    _attr_entity_category = EntityCategory.CONFIG
+    def __init__(
+        self,
+        robot: _RobotT,
+        hub: LitterRobotHub,
+        description: RobotButtonEntityDescription[_RobotT],
+    ) -> None:
+        """Initialize a Litter-Robot button entity."""
+        assert description.name
+        super().__init__(robot, description.name, hub)
+        self.entity_description = description
 
     async def async_press(self) -> None:
         """Press the button."""
-        await self.robot.reset_waste_drawer()
-        self.coordinator.async_set_updated_data(True)
+        if action := self.entity_description.press_fn(self.robot):
+            await action
+            self.coordinator.async_set_updated_data(True)
```

Comment:
This can be reduced
```suggestion
        await self.entity_description.press_fn(self.robot)
        self.coordinator.async_set_updated_data(True)
```